yarn-project/end-to-end/src/e2e_inclusion_proofs_contract.test.ts

```diff
@@ -21,142 +31,218 @@ describe('e2e_inclusion_proofs_contract', () => {
+      }
+
+      {
+        // We test the failure case now --> The proof should fail when the nullifier already exists
+        const receipt = await contract.methods.nullify_note(owner).send().wait({ debug: true });
+        const { newNullifiers } = receipt.debugInfo!;
+        expect(newNullifiers.length).toBe(2);
+
+        const blockNumber = await pxe.getBlockNumber();
+        const nullifier = newNullifiers[1];
+        // Note: getLowNullifierMembershipWitness returns the membership witness of the nullifier itself and not
+        // the low nullifier when the nullifier already exists in the tree and for this reason the execution fails
+        // on low_nullifier.value < nullifier.value check.
+        await expect(
+          contract.methods.test_nullifier_non_inclusion_proof(owner, blockNumber, nullifier).send().wait(),
+        ).rejects.toThrowError(
+          /Proving nullifier non-inclusion failed: low_nullifier.value < nullifier.value check failed/,
+        );
+      }
+    });
+
+    it('proves note validity (note commitment inclusion and nullifier non-inclusion)', async () => {
+      // Owner of a note
+      const owner = accounts[0].address;
+      let noteCreationBlockNumber: number;
+      {
+        // Create a note
+        const value = 100n;
+        const receipt = await contract.methods.create_note(owner, value).send().wait({ debug: true });
+
+        noteCreationBlockNumber = receipt.blockNumber!;
+        const { newCommitments, visibleNotes } = receipt.debugInfo!;
+
+        expect(newCommitments.length).toBe(1);
+        expect(visibleNotes.length).toBe(1);
+        const [receivedValue, receivedOwner, _randomness] = visibleNotes[0].note.items;
+        expect(receivedValue.toBigInt()).toBe(value);
+        expect(receivedOwner).toEqual(owner.toField());
+      }
+
+      {
+        // Prove note validity
+        await contract.methods.test_note_validity_proof(owner, noteCreationBlockNumber).send().wait();
+      }
+    });
+
+    it('note existence failure case', async () => {
+      // Owner of a note - ignored in the contract since the note won't be found and the spare random note commitment
+      // will be used instead
+      const owner = AztecAddress.random();
+
+      // Choose random block number between deployment and current block number to test archival node
+      const blockNumber = await getRandomBlockNumberSinceDeployment();
+      const randomNoteCommitment = Fr.random();
       await expect(
-        contract.methods.test_nullifier_non_inclusion_proof(owner, blockNumber, nullifier).send().wait(),
-      ).rejects.toThrowError(
-        /Proving nullifier non-inclusion failed: low_nullifier.value < nullifier.value check failed/,
-      );
-    }
+        contract.methods.test_note_inclusion_proof(owner, blockNumber, randomNoteCommitment).send().wait(),
+      ).rejects.toThrow(/Leaf value: 0x[0-9a-fA-F]+ not found in NOTE_HASH_TREE/);
```

Comment:
Did not knew you could do this, personally think I prefer testing strictly against an exact value though. But think its fine for now, not what we are changing here.

Reply:
Agree that exact values are nicer. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/3680/commits/c6483ae49bfd5c91e1d80514c3438b3010dc45e5